src/Microsoft.AspNetCore.Mvc.TagHelpers/FormTagHelper.cs

```diff
@@ -173,9 +173,29 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                             RouteValuesPrefix));
                 }
 
-                // User is using the FormTagHelper like a normal <form> tag. Antiforgery default should be false to
-                // not force the antiforgery token on the user.
-                antiforgeryDefault = false;
+                string attributeValue = null;
+                switch (actionAttribute.Value)
+                {
+                    case HtmlString htmlString:
+                        attributeValue = htmlString.ToString();
+                        break;
+                    case string stringValue:
+                        attributeValue = stringValue;
+                        break;
+                }
+
+                if (string.IsNullOrEmpty(attributeValue))
+                {
+                    // User is using the FormTagHelper like a normal <form> tag that has an empty action attribute.
+                    // i.e. <form action="" method="post">
+                    antiforgeryDefault = true;
```

Comment:
Shoot. It actually returns a plain `string`. I thought (hoped) it avoided concatenation and returned a more-segmented `IHtmlContent`.

Just mention the non-`string` / non-`HtmlString` possibility. It's not necessarily an empty `string` or `HtmlString.Empty`.